### DISCUSSION TITLE
FreeRADIUS Do not update user offline status

### DISCUSSION THREAD

**Discussion by TelDragon** [strongswan/strongswan#2405] [Q&A]
Dear author, hello.
I have a question to ask.

I am currently following the previous guidelines and using FreeRadius+Strongswan.

The problem I am currently facing is. When I connect and disconnect normally between Windows and Android app clients, I exit
FreeRadius will update the user status flag as offline or offline. I analyzed that it was because the client sent the offline package correctly.
When working as a Linux client. The following situations may occur. For example, Linux disconnects from the network (network connection is not available). Or simply shut down for other reasons.
FreeRADIUS   The user will still be displayed as online.
How should this situation be changed or optimized?

**Comment by strongX509** [comment]
On a Linux system a network interruption does not terminate an IKE or IPsec SA until the next rekeying time is up. If you want to disconnect quickly you have to activate Dead Peer Detection (DPD) which will discontinue the IKE SA and all depending IPsec SAs after several unacknowledged keep-alive IKE messages.

**Reply by TelDragon** [reply]
Thank you for your reply.
When the network is abnormal or shut down, it cannot send normally
```
Acct-Status-Type = Stop
```
And the next time the connection is restored, radius will only update the time.

So how should this dpd be used?

My dpd is currently set to restart
```
connections {
    eap-linux {
        version = 2
        proposals = default
        remote_addrs = control.xxx.com.cn
        vips = 0.0.0.0
        local {
            auth = eap
            eap_id = xxxxx
        }
        remote {
            auth = pubkey
            id = control.xxx.com.cn 
        }
        children {
            eap-linux {
                start_action = start
                dpd_action = restart
                remote_ts = 11.11.11.0/24
                local_ts = dynamic
                esp_proposals = default
            }
        } 
    }
}
secrets {
   eap-xxxxx {
      id = xxxxxxxx
      secret = xxxxxx
   }
}
```


Do you want to change the server configuration to stop?

```
dpd_action = stop
```
